asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py

```diff
@@ -110,12 +115,14 @@ def __init__(
             else:
                 self.logger.warning(f"Pose {pose} does not exist, skipping.")
 
-        if not protein.exists():
-            raise ValueError(f"Protein {protein} does not exist.")
-
-        self.protein = openeye_perceive_residues(
-            load_openeye_pdb(str(protein)), preserve_all=True
-        )
+        if isinstance(protein, oechem.OEMolBase):
+            self.protein = protein.CreateCopy()
+        else:
+            if not protein.exists():
+                raise ValueError(f"Protein {protein} does not exist.")
+            self.protein = openeye_perceive_residues(
+                load_openeye_pdb(str(protein)), preserve_all=True
+            )
```

Comment:
same things here
1) why not use Target schema?
2) if `protein` can be an `oechem.OEMolBase`, that should be reflected in the type hint

Reply:
Fixed type hint, we will fix up this in a v2.